### PR TITLE
Fixed build warnings in src/Nancy

### PR DESCRIPTION
**Pull request by thecodejunkie**
### Prerequisites

- [X] I have written a descriptive pull-request title
- [X] I have verified that there are no overlapping [pull-requests](https://github.com/NancyFx/Nancy/pulls) open
- [X] I have verified that I am following the Nancy [code style guidelines](https://github.com/NancyFx/Nancy/blob/45238076ad0b7f6ecabd6bae8469e30458d02efe/CONTRIBUTING.md#style-guidelines)
- [X] I have provided test coverage for my change (where applicable)

### Description
Fixed build warnings. The following was changed

- Fixed invalid XML comments
- Added missing XML comments
- Added a couple of missing `this.`
- Changes a couple of `String` to `string`
- Wrapped a single line `if`-statement in curly-braces

The one code change that went in was in [DiagnosticsHook](https://github.com/NancyFx/Nancy/pull/2667/files#diff-5f26f534f32bbe24db978e4c188e3206R191) where we don't really do async handling and we were calling on a method that returns `Task`. I've changed that to `Wait()` on the task

### CODE REVIEW

**Comment by thecodejunkie** [comment]
👀 @NancyFx/most-valued-minions

File: src/Nancy/Diagnostics/DiagnosticsHook.cs [hunk 1]
```diff
@@ -188,7 +188,7 @@ private static Response ExecuteDiagnostics(NancyContext ctx, IRouteResolver rout
 
             if (resolveResult.After != null)
             {
-                resolveResult.After.Invoke(ctx, CancellationToken);
+                resolveResult.After.Invoke(ctx, CancellationToken).Wait();
```
Comment (khellang) [inline src/Nancy/Diagnostics/DiagnosticsHook.cs]: This isn't good 😢

Comment (thecodejunkie) [inline src/Nancy/Diagnostics/DiagnosticsHook.cs]: Yeah I know and I had a chat with @damianh about it.. the real alternative would be to convert the diagnostics stuff completely to async as well I guess?

Comment (khellang) [inline src/Nancy/Diagnostics/DiagnosticsHook.cs]: Yeah, we must not forget that we have a pretty half-assed async story right now. I think we really need to look at getting everything async soon -- before we release 2.0.

Comment (thecodejunkie) [inline src/Nancy/Diagnostics/DiagnosticsHook.cs]: Yeah just gave a nudge at #2577

Comment (thecodejunkie) [inline src/Nancy/Diagnostics/DiagnosticsHook.cs]: I can revert the `Wait()` so we can get the other stuff in?

Comment (khellang) [inline src/Nancy/Diagnostics/DiagnosticsHook.cs]: Yeah, let's tackle everything when we're ready for it 😝

Comment (damianh) [inline src/Nancy/Diagnostics/DiagnosticsHook.cs]: This is one of those changes that will have a knock on effect and will explode this PR. I would agree to merge this PR as is and fix this `.Wait()` in a subsequent PR.

**Comment by thecodejunkie** [comment]
@khellang @damianh reverted the `Wait()`

**Comment by thecodejunkie** [comment]
We're actually going a `Wait()` on line 179 (added by me apparently, during the new route syntax stuff). Should I nuke that as well?

**Comment by khellang** [comment]
Yeah, just nuke it while you're there :smile:

**Comment by khellang** [comment]
Argh. The damn `Should_return_JSON_serialized_form` test is failing on Travis again. Or did we never get to fix it?

**Comment by thecodejunkie** [comment]
@khellang removed the second `Wait()`. We resolved `Should_return_JSON_serialized_form` by upgrading the used Mono version on Travis =/

**Pull request merged**